pulsar-broker-common/src/main/java/com/yahoo/pulsar/broker/ServiceConfiguration.java

```diff
@@ -291,12 +301,32 @@ public void setWebServicePortTls(int webServicePortTls) {
         this.webServicePortTls = webServicePortTls;
     }
 
-    public boolean isBindOnLocalhost() {
-        return bindOnLocalhost;
+    public String getBindAddress() {
+        if (this.bindAddress == null) {
+            try {
+                LOG.debug("bindAddress not set, attempting default configuration.");
+                this.setBindAddress(InetAddress.getLocalHost().getHostName());
+            } catch (UnknownHostException ex) {
+                LOG.warn(ex.getMessage(), "Using localhost as bindAddress.");
+                this.setAdvertisedAddress("localhost");
+            }
+        }
+        return this.bindAddress;
+    }
+
+    public void setBindAddress(String bindAddress) {
+        this.bindAddress = bindAddress;
+    }
+
+    public String getAdvertisedAddress() {
```

Comment:
Same as above, we should just return what was configured, and in any case, the fallback here should be on `hostname` rather than the bind address.